### DISCUSSION TITLE
Image Header Text Extraction Issues

### DISCUSSION THREAD

**Discussion by asnyd257** [ImageMagick/ImageMagick#6818] [Help]
### ImageMagick version

7.1.1-Q16-HDRI

### Operating system, version and so on

Windows 11 Pro

### Description

Good afternoon! I'm hoping someone here can give me some direction on an issue I'm facing.

So, a bit of a preface...I'm not asking for someone to write anything for me, merely some advice on what to look into. Although of course if anyone has existing commands that'd do something similar as a starting point like that is often extremely helpful. I'm new to image manipulation and OCRs in general but I'm a programmer by trade so I can typically understand fairly technical directions.

Now, here's my basic issue. I've written code to capture a screencap of a particular window I'm interested in extracting text from - it's a UI with a lot of little boxes/headers containing text I need to view for the rest of my program to function properly. The "body" text of the program (for lack of a better term) I can scan fine but it's the headers that are proving problematic, I suspect due to the coloring of the initial window before I convert it greyscale.

Original UI window (ignore the red boxes, those are just blocking sensitive info):
![tesseract2](https://github.com/ImageMagick/ImageMagick/assets/110253304/f3152fd5-3a53-4b31-bd3c-033266867aeb)

Greyscale version (same thing, ignore the boxes - I left some of the text I'm able to extract exposed to illustrate):
![tesseract1](https://github.com/ImageMagick/ImageMagick/assets/110253304/3c9e390e-ce8d-490f-8b71-4e39db1e8b97)

The red arrows pointing to the headers are the problem areas - the green arrow is pointing at what I'd call "body" text and I can extract that with decent accuracy. I can't binarize using the methods I've learned about or I lose the headers entirely. Can someone point me towards some potential solutions for a method to preprocess this image that is able to make the headings "visible" to the OCR I'm using? That's my current sticking point. And, if it's relevant, I'm using Tesseract which I'm also new to.

Thank you in advance for any assistance you can give in this issue.

**Comment by snibgo** [comment]
https://imagemagick.org/script/command-line-options.php#level-colors is a reasonable start. This converts a certain colour to black, another certain colour to white, and any other colours to something sensible (interpolated).
```
magick in.jpg -level-colors sRGB(23%,59%,87%),sRGB(0,23%,75%) out.png
```
I have chosen to convert the light blue background to black, and the dark blue letters to white.

And my usual caution: never start from JPG inputs unless you really have to. If you are doing the screengrabs, then you certainly don't have use JPG.

**Reply by asnyd257** [reply]
Thank you! I'll begin looking into what you suggested. And no worries, my source images are .png - the jpgs are just from when I blocked out the sensitive info.

Much appreciated!
